File: python/2010/scott-pilgrim-vs-the-world-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add Scott Pilgrim vs. the World to the database
movies.insert(
    title="Scott Pilgrim vs. the World", 
    year=2010, 
    plot="Scott Pilgrim must defeat his new girlfriend's seven evil exes in order to win her heart.", 
    rating=7.5
)

# Confirm that the movie was added
movie = movies.select(
    title="Scott Pilgrim vs. the World", 
    year=2010
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    

File: csharp/2010/scott-pilgrim-vs-the-world-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "Scott Pilgrim vs. the World" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "Scott Pilgrim vs. the World",
                year: 2010,
                plot: "Scott Pilgrim must defeat his new girlfriend's seven evil exes in order to win her heart.",
                rating: 7.5
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "Scott Pilgrim vs. the World",
                year: 2010
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}
